tests/unit/constraints/test_tabular.py

```diff
@@ -787,1789 +787,283 @@ def test_reverse_transform_non_string(self):
 
         Input:
-        - Table with values above and below `low`.
-        Output:
-        - True should be returned for the rows where the high
-        column is above `low`.
+        - low = 'a'
+        - high = 'b'
+        - strict_boundaries = True
         """
-        # Setup
-        low_dt = pd.to_datetime('8/31/2021')
-        instance = GreaterThan(low=low_dt, high='a', strict=False, scalar='low')
-        table_data = pd.DataFrame({
-            'a': [datetime(2021, 9, 17), datetime(2021, 7, 1), datetime(2021, 9, 1)],
-            'b': [4, 2, 2],
-        })
-
         # Run
-        out = instance.is_valid(table_data)
+        instance = Inequality(low_column_name='a', high_column_name='b', strict_boundaries=True)
 
         # Assert
-        expected_out = [True, False, True]
-        np.testing.assert_array_equal(expected_out, out)
+        assert instance._operator == np.greater
 
-    def test_is_valid_two_cols_with_nans(self):
-        """Test the ``GreaterThan.is_valid`` method with nan values.
+    def test__validate_columns_exist_incorrect_columns(self):
+        """Test the ``Inequality._validate_columns_exist`` method.
 
-        If there is a NaN row, expect that `is_valid` returns True.
+        This method raises an error if ``low_column_name`` or ``high_column_name`` do not exist.
 
         Input:
-        - Table with a NaN row
-        Output:
-        - True should be returned for the NaN row.
+        - Table with given data.
         """
         # Setup
-        instance = GreaterThan(low='a', high='b', strict=True)
-
-        # Run
         table_data = pd.DataFrame({
-            'a': [1, None, 3],
-            'b': [4, None, 2],
-            'c': [7, 8, 9]
+            'a': [1, 2, 4],
+            'b': [4, 5, 6]
         })
-        out = instance.is_valid(table_data)
+        instance = Inequality(low_column_name='a', high_column_name='c')
 
-        # Assert
-        expected_out = [True, True, False]
-        np.testing.assert_array_equal(expected_out, out)
+        # Run / Assert
+        with pytest.raises(KeyError):
+            instance._validate_columns_exist(table_data)
 
-    def test_is_valid_two_cols_with_one_nan(self):
-        """Test the ``GreaterThan.is_valid`` method with nan values.
+    def test__fit(self):
+        """Test the ``Inequality._fit`` method.
 
-        If there is a row in which we compare one NaN value with one
-        non-NaN value, expect that `is_valid` returns True.
+        The method should learn the ``dtype`` of ``_column_name`` and ``is_datetime``.
 
         Input:
-        - Table with a row that contains only one NaN value.
-        Output:
-        - True should be returned for the row with the NaN value.
+        - Table data with integers.
+        Side Effect:
+        - _validate_columns_exist should be called once
+        - _get_is_datetime should be called once
+        - _is_datetime should receive the output of _get_is_datetime
```

Comment:
Does `is_datetime` have the preceding `_` or not, or are there two different variables?